cpp/include/Ice/ClientAuthenticationOptions.h

```diff
@@ -0,0 +1,414 @@
+         *
+         * Example of setting clientCertificateSelectionCallback:
+         * ```cpp
+         * PCCERT_CONTEXT _clientCertificate = ...; // Load the client certificate using WinCrypt API
+         *
+         * auto initData = Ice::InitializationData {
+         *   ...
+         *   .clientAuthenticationOptions = ClientAuthenticationOptions {
+         *      .clientCertificateSelectionCallback = [this](const std::string&)
+         *      {
+         *        // Increment the certificate context reference count to ensure it remains
+         *        // valid for the duration of the connection. The SSL transport will release
+         *        // it after closing the connection.
+         *        CertDuplicateCertificateContext(_clientCertificate);
+         *        return _clientCertificate;
+         *      }
+         * };
+         *
+         * auto communicator = Ice::initialize(initData);
+         * ...
+         * CertFreeCertificateContext(_clientCertificate); // Release the certificate when no longer needed
+         * ```
+         *
+         * See Detailed Wincrypt documentation for [CERT_CONTEXT](
+         * https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/ns-wincrypt-cert_context) and
+         * [CertCreateCertificateContext](https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certcreatecertificatecontext)
+         */
+        std::function<PCCERT_CONTEXT(const std::string& host)> clientCertificateSelectionCallback;
+
+        /**
+         * The trusted root certificates used for validating the server's certificate chain. If this field is set, the
+         * server's certificate chain is validated against these certificates; otherwise, the system's default root
+         * certificates are used.
+         *
+         * @remarks The trusted root certificates are used by both the default validation callback, and by custom
+         * validation callback set in clientCertificateValidationCallback.
+         *
+         * The application must ensure that the certificate store remains valid during the setup of the Communicator. It
+         * is also the application's responsibility to release the certificate store object after the Communicator has
+         * been created to prevent memory leaks.
+         *
+         * Example of setting trustedRootCertificates:
+         * ```cpp
+         * HCERTSTORE rootCerts = ...; // Populate with X.509 certificates
+         *
+         * auto initData = Ice::InitializationData {
+         *   ...
+         *   .clientAuthenticationOptions = ClientAuthenticationOptions {
+         *      .trustedRootCertificates = rootCerts;
+         *   }
+         * };
+         *
+         * auto communicator = Ice::initialize(initData);
+         * CertCloseStore(rootCerts); // It is safe to close the rootCerts store now.
+         * ```
+         */
+        HCERTSTORE trustedRootCertificates;
```

Comment:
Should we set a default, like `= nullptr`. Otherwise, how do we know it was set?